### PR TITLE
BAVL-342 adding basic integration test for video booking migration.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
PR to add some basic integration test coverage for the video booking migration process.

Even though there are 17 modified files, the additional changes on this PR are really small/nominal changes.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/prisonersearch/PrisonerSearchClient.kt [hunk 1]
```diff
@@ -21,15 +21,6 @@ class PrisonerSearchClient(private val prisonerSearchApiWebClient: WebClient) {
       .bodyToMono(Prisoner::class.java)
       .onErrorResume(WebClientResponseException.NotFound::class.java) { Mono.empty() }
       .block()
-
-  fun getPrisoner(bookingId: Long): Prisoner? =
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/prisonersearch/PrisonerSearchClient.kt:25]: Moved function into migration client as only really needed there.  When migration is done the migration client will be deleted.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/SqsIntegrationTestBase.kt [hunk 2]
```diff
@@ -30,7 +27,8 @@ import java.util.concurrent.TimeUnit
 @ActiveProfiles("test-localstack", inheritProfiles = false)
 abstract class SqsIntegrationTestBase : IntegrationTestBase() {
 
-  private val mapper: ObjectMapper = jacksonObjectMapper().registerModule(JavaTimeModule()).configure(DeserializationFeature.FAIL_ON_IGNORED_PROPERTIES, false)
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/SqsIntegrationTestBase.kt:33]: Don't know why I did this originally, should be using Springs autowired mapper.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/MigrateBookingIntegrationTest.kt [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+
+    with(migratedCourtBooking) {
+      bookingType isEqualTo "COURT"
+      statusCode isEqualTo StatusCode.ACTIVE
+      hearingType isEqualTo "UNKNOWN"
+      createdBy isEqualTo "MIGRATION COURT USER"
+      createdTime isCloseTo LocalDateTime.now()
+      createdByPrison isBool false
+      migratedVideoBookingId isEqualTo 1
+      // TODO if there is an update/amend should the amended attrs be set?
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/MigrateBookingIntegrationTest.kt:117]: Thoughts?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/MigrateBookingIntegrationTest.kt [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+
+    with(migratedProbationBooking) {
+      bookingType isEqualTo "PROBATION"
+      statusCode isEqualTo StatusCode.ACTIVE
+      probationMeetingType isEqualTo "UNKNOWN"
+      createdBy isEqualTo "MIGRATION PROBATION USER"
+      createdTime isCloseTo LocalDateTime.now()
+      createdByPrison isBool false
+      migratedVideoBookingId isEqualTo 2
+      // TODO if there is an update/amend should the amended attrs be set?
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/MigrateBookingIntegrationTest.kt:263]: Thoughts anyone?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/MigrateBookingIntegrationTest.kt [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+    inboundEventsService.process(MigrateVideoBookingEvent(1))
+
+    val migratedCourtBooking = videoBookingRepository.findAll().single(VideoBooking::isCourtBooking)
+
+    with(migratedCourtBooking) {
+      bookingType isEqualTo "COURT"
+      statusCode isEqualTo StatusCode.ACTIVE
+      hearingType isEqualTo "UNKNOWN"
+      createdBy isEqualTo "MIGRATION COURT USER"
+      createdTime isCloseTo LocalDateTime.now()
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/MigrateBookingIntegrationTest.kt:114]: Is `createdTime` right or should it be set to when the original booking was first created?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/MigrateBookingIntegrationTest.kt [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+    videoBookingRepository.findAll().filter(VideoBooking::isCourtBooking) hasSize 0
+
+    inboundEventsService.process(MigrateVideoBookingEvent(1))
+
+    val migratedCourtBooking = videoBookingRepository.findAll().single(VideoBooking::isCourtBooking)
+
+    with(migratedCourtBooking) {
+      bookingType isEqualTo "COURT"
+      statusCode isEqualTo StatusCode.ACTIVE
+      hearingType isEqualTo "UNKNOWN"
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/MigrateBookingIntegrationTest.kt:112]: Still need to sort out the data to back up 'UNKNOWN' if that is what we are going with.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/MigrateBookingIntegrationTest.kt [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+    videoBookingRepository.findAll().filter(VideoBooking::isProbationBooking) hasSize 0
+
+    inboundEventsService.process(MigrateVideoBookingEvent(2))
+
+    val migratedProbationBooking = videoBookingRepository.findAll().single(VideoBooking::isProbationBooking)
+
+    with(migratedProbationBooking) {
+      bookingType isEqualTo "PROBATION"
+      statusCode isEqualTo StatusCode.ACTIVE
+      probationMeetingType isEqualTo "UNKNOWN"
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/MigrateBookingIntegrationTest.kt:258]: Still need to sort out the data to back up 'UNKNOWN' if that is what we are going with.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/migration/MigrateVideoBookingServiceTest.kt [hunk 8]
```diff
@@ -668,6 +668,114 @@ class MigrateVideoBookingServiceTest {
...
+      createdTime isEqualTo 2.daysAgo().atStartOfDay()
+
+      appointments().single()
+        .isAtPrison(WERRINGTON)
+        .isForPrisonerNumber("DEF123")
+        .isOnDate(yesterday())
+        .startsAt(LocalTime.of(10, 0))
+        .endsAt(LocalTime.of(11, 0))
+        .isForAppointmentType("VLB_PROBATION")
+        .isAtLocation(werringtonLocation)
```
Comment (timharrison-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/migration/MigrateVideoBookingServiceTest.kt:760]: Not sure about the appointment type?  Even probation appointments should still be of type `Video Link - Court hearing` ("VLB")  until we put in specific probation journeys later?

**Review by timharrison-moj — COMMENTED** [review]
LGTM - though the appointment type question is an important one, so just commenting for now.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/MigrateBookingIntegrationTest.kt (hunk 5 above)
Comment (timharrison-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/MigrateBookingIntegrationTest.kt:114]: I suspect the original time would be best - for reporting/ extracts.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/MigrateBookingIntegrationTest.kt (hunk 3 above)
Comment (timharrison-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/MigrateBookingIntegrationTest.kt:117]: Yes - I think maybe the lastUpdatedBy and lastUpdatedTime should probably reflect the most recent change (event)

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/MigrateBookingIntegrationTest.kt (hunk 4 above)
Comment (timharrison-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/MigrateBookingIntegrationTest.kt:263]: Yes please. I think so.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/migration/MigrateVideoBookingServiceTest.kt (hunk 8 above)
Comment (timharrison-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/migration/MigrateVideoBookingServiceTest.kt:760]: I take it back. Correct as is.

**Review by timharrison-moj — APPROVED** [review]
LGTM.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/prisonersearch/PrisonerSearchClient.kt (hunk 1 above)
Comment (matt-crowson-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/prisonersearch/PrisonerSearchClient.kt:25]: I am trying to group everything that is migration related only into migration specific classes.

This should make the cleanup/removal of migration **only** code easier i.e. touching other classes as little as possible so unused code does not get left in.

**Pull request merged**